### ISSUE TITLE
Unable to see whole path when creating file with very long path/name and dashes

### ISSUE THREAD

**Issue by adamreisnz** [atom/tree-view#473]
Try creating a new file (or duplicating/renaming one), and type a very long name into the dialog box (e.g. some/path/my-very-very-long-filename-to-test-this-issue).

You will notice that once the text reaches the end of the box, you cannot see the end of the filename anymore. Moving left/right/up/down with arrows doesn't help. Only pressing space seems to trigger the display of the end of the filename.

**Comment by mnquintana** [comment]
Would you mind attaching a screen capture of this behavior? I can't seem to reproduce this with OS X 10.10.3 and Atom 0.208.0-c06d62e, but I want to make sure I'm looking for the right thing.

**Comment by AdamBuczynski** [comment]
Yes, sure thing. Can you recommend a decent screen grabber? I tried Gifgrabber earlier, but it was rather obnoxious...

**Comment by mnquintana** [comment]
@AdamBuczynski You can try [LICEcap](http://www.cockos.com/licecap/) (which is what I usually use)

**Comment by AdamBuczynski** [comment]
![Screengrab](https://www.dropbox.com/s/j0scb4o5l1e0k76/long-filename.gif?dl=1)

**Comment by williamboman** [comment]
+1. @adambuczynski Have you solved this somehow?

**Comment by adambuczynski** [comment]
Sorry no I haven't.
@mnquintana could you update the status from more info needed to bug or do you still need more information?

**Comment by williamboman** [comment]
@adambuczynski Which themes do you use?

**Comment by adambuczynski** [comment]
That screen cap was with the `Atom Dark` theme (comes bundled with Atom).

**Comment by adambuczynski** [comment]
In the `One Dark` theme, the input box turns to multi line, which is a neat solution:

![ex](https://www.dropbox.com/s/e1ag0yds4s4htnd/Screenshot%202015-06-18%2010.17.29.png?dl=1)

Maybe the `Atom Dark` theme can adopt this as well. Is there a issues repository for that specific theme?

**Comment by izuzak** [comment]
cc @simurai for :art: :eyes:

**Comment by simurai** [comment]
I'm not able to reproduce it. In my case, the file name just keeps moving to the left:

![new-file](https://cloud.githubusercontent.com/assets/378023/8266271/c47c6726-1760-11e5-8b01-6146a17f14fe.gif)

Same thing with `One dark`.

> As you can notice, when the filename becomes too long, without spaces/slashes, it doesn't move the view to the "right" to see where you're typing. When using spaces, it does.

It sounds like the text wraps to a new line. Have you tried with `atom --safe` to only test with core packages?

**Comment by adambuczynski** [comment]
Same happening with `atom --safe`

![ex](https://www.dropbox.com/s/bvhju0oz1215y7a/with%20atom%20safe.gif?dl=1)

Could you try renaming a file with slashes in the path to see if that makes a difference?

**Comment by izuzak** [comment]
@simurai :open_mouth: that's strange -- I can definitely reproduce this in the latest master version of Atom (0.211.0-fc4f2a1) in safe mode on OSX 10.10.3 with the Atom Dark theme:

![add](https://cloud.githubusercontent.com/assets/38924/8266598/52d9ad6e-1737-11e5-8c9a-bc147059f032.gif)

**Comment by simurai** [comment]
Haa.. me still not and I got the exact same setup as @izuzak.

![new-file](https://cloud.githubusercontent.com/assets/378023/8266625/4d23f7a8-1773-11e5-8bc2-1182ee939763.gif)

And just to be sure, what syntax theme? Although I tried all of the core ones. Also retina/non-retina screen.

**Comment by izuzak** [comment]
Ah man, now it's getting interesting :D I think I see the problem -- I see the same behavior as @simurai if I turn the Soft Wrap config setting OFF, but if I have Soft Wrap turned ON -- I see the behavior described in the issue. Do you see the same, @simurai?

(Syntax theme: Atom dark, UI theme: Atom dark)

**Comment by simurai** [comment]
Yes, that's it. Kinda makes sense now. Seems the Soft Wrap settings also works for mini editors. Which it shouldn't, I guess.

That's something I wondered already a few times. Wouldn't it be simpler to just use `<input>`s? Or what's the reason that all inputs (like here or in the settings) are full featured `atom-text-editor`s?

**Comment by izuzak** [comment]
> Or what's the reason that all inputs (like here or in the settings) are full featured atom-text-editors?

Wish I could answer that question :smile: 

However, the interesting thing here is that the OneDark/OneDark theme combo behaves differently than the AtomDark/AtomDark theme combo with the Soft Wrap config setting turned ON -- OneDark/OneDark expands the textbox while the AtomDark/AtomDark doesn't. With the Soft Wrap config setting turned ON -- both theme combos have the same behavior: the textbox scrolls horizontally as you type.

**Comment by simurai** [comment]
Ya, the mini editors in the One themes don't have a fixed height and just expand if the content grows. I think it wasn't intended like this and is just a lucky side effect of making the UI resizable based on font size.

I guess we could try the same with Atom dark/light. Although other themes will still have this issue. So it might also be worth preventing soft wrap for all mini editors.. except find+replace, there it's probably needed.

**Comment by acusti** [comment]
Should this issue be in a different repo? It isn’t specific the tree view, but rather applies to any “mini” `atom-text-editor`. I opened a duplicate issue in atom/atom#7228, but then closed it when I realized it was a duplicate; maybe keeping it there makes more sense, considering that the behavior lives in [that repo](https://github.com/atom/atom/blob/master/src/text-editor.coffee).

From my perspective, while the One theme behaviour is an interesting side effect, I think the best solution certainly is to disable soft wrap on all mini `atom-text-editors`. They look like text inputs, so I think user expectation is that they will behave like text inputs. Looking at the `TextEditor` model, the `ignoreInvisibles` flag already has special handling ([here](https://github.com/atom/atom/blob/master/src/text-editor.coffee#L87) and [here](https://github.com/atom/atom/blob/master/src/text-editor.coffee#L488)) based on the `@mini` attribute, so extending those exceptions to include modifying the `softWrapped` attribute (which is also set on the `DisplayBuffer`) seems pretty straightforward.

I guess tests for `softWrapped` would then need to go in the `text-editor-spec`, though that spec seems pretty low level. Should they instead be part of the [“when the 'mini' property is true” suite](https://github.com/atom/atom/blob/master/spec/text-editor-component-spec.coffee#L2739) in the `text-editor-component-spec`?

Would it be worthwhile at the same time to extend the text editor [`instantiation` tests](https://github.com/atom/atom/blob/master/spec/text-editor-element-spec.coffee#L12) to check if the constructor sets `ignoreInvisibles` and `softWrapped` based on the value of the `mini` attribute? And to add a separate test of the same behavior for the `setMini` method?

**Comment by louisgv** [comment]
Hi all, 

How can I temporary fix this behavior? I need soft wrap for coding but also want the horizontal behavior for the mini path editor.

Interestingly, Softwrap was disabled but still work with markdown files:

![screenshot from 2015-08-13 01 47 36](https://cloud.githubusercontent.com/assets/6723574/9246043/b9731b46-415d-11e5-87eb-45b8f0003f96.png)


I'm using Seti UI theme and Monokai Syntax

**Comment by simurai** [comment]
> They look like text inputs, so I think user expectation is that they will behave like text inputs.

I kinda agree with this. Having the input expand to multiline might work in some cases, but could also cause layout issues.

> I opened a duplicate issue in atom/atom#7228, but then closed it when I realized it was a duplicate; maybe keeping it there makes more sense, considering that the behavior lives in that repo.

Yes, now that the root cause (SoftWrap) is known, it would make sense. Maybe issue https://github.com/atom/atom/issues/5503 will take over. Proposal is to have `<atom-text-editor>`, `<atom-text-area>` and `<atom-text-input>` (current mini editor).

**Comment by simurai** [comment]
As a short term fix, you can add

```css
atom-text-editor[mini] {
  height: auto;
  max-height: none;
}
```

to your `styles.less`. It makes the height of mini editors expandable.

![screen shot 2015-08-15 at 11 33 42 pm](https://cloud.githubusercontent.com/assets/378023/9289304/393d4c26-43a6-11e5-8636-37c4d5f7fde7.png)

Should work with most themes.

**Comment by Trudko** [comment]
+1 thanks @simurai for workaround.

**Comment by swrobel** [comment]
Using the recommended styles.less fix isn't working for me. I tested on multiple themes in Atom 1.2.4 on OS X 10.11. Is there any hope of getting this fixed soon?

**Comment by arielnoname** [comment]
![naming](https://cloud.githubusercontent.com/assets/11996305/11693874/0348d528-9e86-11e5-8881-aa9edc22e111.gif)

For the record.
Now: OSX 10.11, Atom 1.2.4
Happened to me since always.

Cheers

**Comment by matheussampaio** [comment]
I'm still having this issue, even in safe mode.

## Versions

Atom : 1.8.0
Node : 6.2.2
OSX: 10.11.5

**Comment by clemsos** [comment]
Idem here on Debian with Atom 1.2.1 
![tooltip_161](https://cloud.githubusercontent.com/assets/743246/16945555/c6e027b0-4da5-11e6-812e-455666e475f6.png)

**Comment by viion** [comment]
This is still an issue in Atom 1.8. With the option **Soft Wrap** turned on the text goes to a new line. Turning off the option fixes it. @simurai Solution did not work for me (but could be the Theme).

![screen shot 2016-07-22 at 13 23 16](https://cloud.githubusercontent.com/assets/270800/17057071/bd782cce-500f-11e6-8470-e3261a7e3d93.png)

**Comment by simurai** [comment]
@viion What's the theme you're using? It probably needs a stronger selector.

**Comment by Sam-Hoult** [comment]
I used the fix from @simurai, but added !important flags:
```
atom-text-editor[mini] {
  height: auto !important;
  max-height: none !important;
}
```

**Comment by chiangs** [comment]
@Sam-Hoult , thanks, the !important flags made the difference for me on v 1.12.9 with Seti-ui theme.

**Comment by GioLogist** [comment]
@simurai good temp fix! 

Any plans on getting this incorporated directly into Atom?

**Comment by 50Wliu** [comment]
https://github.com/atom/atom/pull/13764

**Comment by chiangs** [comment]
@viion, I use advanced-open-file package and it works great with wrapping long file paths/names.

**Comment by 50Wliu** [comment]
This will be fixed in Atom 1.17.0.
